src/System.Net.Sockets/src/System/Net/Sockets/Socket.Windows.cs

```diff
@@ -122,14 +122,13 @@ internal unsafe bool TransmitPackets(SafeCloseSocket socketHandle, IntPtr packet
             return transmitPackets(socketHandle, packetArray, elementCount, sendSize, overlapped, flags);
         }
 
-        internal static IntPtr[] SocketListToFileDescriptorSet(IList socketList)
+        internal static void SocketListToFileDescriptorSet(IList socketList, Span<IntPtr> fileDescriptorSet)
         {
             if (socketList == null || socketList.Count == 0)
             {
-                return null;
+                return;
             }
 
-            IntPtr[] fileDescriptorSet = new IntPtr[socketList.Count + 1];
             fileDescriptorSet[0] = (IntPtr)socketList.Count;
             for (int current = 0; current < socketList.Count; current++)
```

Comment:
`fileDescriptorSet` is now the same length as `socketList`, right?  If so, it'd be more efficient to use fileDescriptorSet.Length rather than `socketList.Count`, which is an interface call.  If we need to keep using `socketList.Count` for some reason, we should store it in a local first.

Reply:
@stephentoub  not if we use the "const sized stackalloc" from above - if we do that, it will be different unless we  do an additional slice; I can change it to only access `socketList.Count` once, though?

Reply:
we could also use `foreach` here, if we have the earlier code set the length

Reply:
You could measure to see which is faster (if measurable), using a fixed size and slicing vs accessing socketList.Count once vs just passing in the count to the helper so it's actually accessed once across all helpers.

Reply:
> although: on IList, that's even more allocatey - so pretend I didn't say that

:smile: More "allocatey", and also still two interface calls per element (MoveNext + Current instead of Count + this[])